oem/smc/drive.go

```diff
@@ -0,0 +1,62 @@
+package smc
+
+import (
+	"encoding/json"
+
+	"github.com/stmcginnis/gofish/redfish"
+)
+
+type Drive struct {
+	redfish.Drive
+	Oem DriveOem `json:"Oem"`
+}
+
+type DriveOem struct {
+	Supermicro struct {
+		Temperature             int
+		PercentageDriveLifeUsed int
+		DriveFunctional         bool
+	} `json:"Supermicro"`
+}
+
+type DriveTarget struct {
+	Target     string `json:"target"`
+	ActionInfo string `json:"@Redfish.ActionInfo"`
+}
+
+type DriveActions struct {
+	redfish.DriveActions
+	Oem struct {
+		DriveIndicate    DriveTarget `json:"#Drive.Indicate"`
+		SmcDriveIndicate DriveTarget `json:"#SmcDrive.Indicate"`
+	} `json:"Oem"`
+}
+
+func FromDrive(drive *redfish.Drive) (Drive, error) {
+	var oem DriveOem
+	err := json.Unmarshal(drive.Oem, &oem)
+
+	return Drive{
+		Drive: *drive,
+		Oem:   oem,
+	}, err
+}
+
+func FromDriveActions(da *redfish.DriveActions) (DriveActions, error) {
```

Comment:
Get rid of this and just have this part of the `Drive` unmarshaling in `FromDrive`.